Improve OTEL docs according to customer feedback (#6195)
getsentry/sentry-docs: src/platform-includes/performance/opentelemetry-install/with-java-agent/without-auto-init/java.mdx

> You can download the latest version of the `sentry-opentelemetry-agent-{{ packages.version('sentry.java.opentelemetry-agent') }}.jar` from [MavenCentral](https://search.maven.org/artifact/io.sentry/sentry-opentelemetry-agent). It is also available as `ZIP` containg the `JAR` used on this page on [GitHub](https://github.com/getsentry/sentry-java/releases/).

Reviewer: ```suggestion
You can download the latest version of the `sentry-opentelemetry-agent-{{ packages.version('sentry.java.opentelemetry-agent') }}.jar` from [MavenCentral](https://search.maven.org/artifact/io.sentry/sentry-opentelemetry-agent). It's also available as a `ZIP` containing the `JAR` used on this page on [GitHub](https://github.com/getsentry/sentry-java/releases/).
```
